src/PhotosphereEditor.tsx

```diff
@@ -1,31 +1,63 @@
 import { useState } from "react";
 
 import AddPhotosphere from "./AddPhotosphere.tsx";
-import { VFE } from "./DataStructures.ts";
+import { Photosphere, VFE } from "./DataStructures.ts";
 import PhotosphereViewer from "./PhotosphereViewer.tsx";
 
+/* -----------------------------------------------------------------------
+    Update the Virtual Field Environment with an added Photosphere.
+
+    * Take the initial VFE from parent
+    * If a change has been made to the initialVFE -> updateTrigger === true
+    * Send the newPhotosphere back to parent 
+    * Parent updates the VFE with the newPhotosphere object
+   ----------------------------------------------------------------------- */
+
+// Properties passed down from parent
 interface PhotosphereEditorProps {
-  vfe: VFE;
+  initialVFE: VFE;
+  onUpdateVFE: (updatedVFE: VFE) => void;
 }
 
-function PhotosphereEditor({ vfe }: PhotosphereEditorProps): JSX.Element {
-  //Basic states for each component, its basically just a boolean
+// If an update is triggered, add newPhotosphere, and update VFE
+function PhotosphereEditor({
+  initialVFE,
+  onUpdateVFE,
+}: PhotosphereEditorProps): JSX.Element {
+  // Base states
+  const [vfe, setVFE] = useState<VFE>(initialVFE);
   const [showAddPhotosphere, setShowAddPhotosphere] = useState(false);
-  //Create a useState for your component
+  const [updateTrigger, setUpdateTrigger] = useState(0);
+
+  // Update the VFE
+  function handleAddPhotosphere(newPhotosphere: Photosphere) {
+    const updatedVFE: VFE = {
+      ...vfe,
+      photospheres: {
+        ...vfe.photospheres,
+        [newPhotosphere.id]: newPhotosphere,
+      },
+    };
+    console.log("Updated VFE (local state in PhotosphereEditor):", updatedVFE);
```

Comment:
Not sure how everyone feels but we might want to remove the console.logs before we merge to main